src/classes/SftpSessionHandler.ts

```diff
@@ -94,39 +94,35 @@ export class SftpSessionHandler {
         attrs,
       },
     );
-    this.getCurrentPermanentFileSystem().then((permFileSystem: PermanentFileSystem) => {
-      permFileSystem.getItemType(filePath)
-        .then((fileType) => {
-          switch (fileType) {
-            case fs.constants.S_IFDIR:
-              logger.verbose(
-                'Response: Status (NO_SUCH_FILE)',
-                {
-                  reqId,
-                  code: SFTP_STATUS_CODE.NO_SUCH_FILE,
-                },
-              );
-              this.sftpConnection.status(reqId, SFTP_STATUS_CODE.NO_SUCH_FILE);
-              break;
-            default: {
-              this.openExistingFileHandler(
-                reqId,
-                filePath,
-                flags,
-              );
-              break;
-            }
-          }
-        }).catch((err: unknown) => {
-          logger.debug(err);
-          this.openNewFileHandler(
+    const permanentfileSystem = this.getCurrentPermanentFileSystem();
+    permanentfileSystem.getItemType(filePath).then((fileType) => {
+      switch (fileType) {
+        case fs.constants.S_IFDIR:
+          logger.verbose(
+            'Response: Status (NO_SUCH_FILE)',
+            {
+              reqId,
+              code: SFTP_STATUS_CODE.NO_SUCH_FILE,
+            },
+          );
+          this.sftpConnection.status(reqId, SFTP_STATUS_CODE.NO_SUCH_FILE);
+          break;
+        default: {
+          this.openExistingFileHandler(
```

Comment:
Question, is there a bottleneck to not awaiting this for example and making the handler method async?

Reply:
We can't make the handler methods async because we don't control their signature -- the ssh2 library (which ultimately gets these handlers passed to them) expects handlers to be synchronous, which means they don't handle failed promises.